tests/unit/experimental/pipeline/test_orchestrator.py

```diff
@@ -16,13 +16,76 @@
 from neo4j_genai.experimental.pipeline import Component
 from neo4j_genai.experimental.pipeline.pipeline import Orchestrator, Pipeline, RunStatus
 
+from tests.unit.experimental.pipeline.components import ComponentPassThrough
+
+
+def test_orchestrator_get_component_inputs_from_user_only() -> None:
+    pipe = Pipeline()
+    pipe.add_component(ComponentPassThrough(), "a")
+    pipe.add_component(ComponentPassThrough(), "b")
+    orchestrator = Orchestrator(pipe)
+    data = orchestrator.get_component_inputs("a", {}, {"a": {"value": "text"}})
+    assert data == {"value": "text"}
+    data = orchestrator.get_component_inputs("b", {}, {"a": {"value": "text"}})
+    assert data == {}
+
+
+def test_pipeline_get_component_inputs_from_parent_specific() -> None:
+    pipe = Pipeline()
+    pipe.add_component(ComponentPassThrough(), "a")
+    pipe.add_component(ComponentPassThrough(), "b")
+    pipe.connect("a", "b", input_config={"value": "a.result"})
+
+    # component "a" already run and results stored:
+    pipe._store.add("a", {"result": "a_result"})
+
+    orchestrator = Orchestrator(pipe)
+    data = orchestrator.get_component_inputs(
+        "b", {"value": "a.result"}, {"a": {"value": "text"}}
```

Comment:
What does the argument `{"a": {"value": "text"}}` specify here?

Reply:
This is the pipeline.run input (it's called `data` within the code base, not very explicit I agree)